### PR TITLE
⏰ refresh certain TE elements when significant time change detected

### LINKED ISSUE #3194: Timer page not refreshed when significant time changes
# 🐛 Bug Report
Timer page doesn't reload when significant time changes:

- Overnight
- Time zone changes

## Context

Build/Branch: `develop`
Phone OS: Any
Phone model: Any

## Expected behavior
- The te log should be updated properly

## To reproduce
- Change the timezone of the phone to Honolulu
- Create some time entries 
- Change the timezone to Brisbane

### PR DESCRIPTION
**Pull request by khoi**
### Relationships

Closes #3194
Supercedes #3195

## Why do we want this?
Just like how we reload the main log every midnight, we might want to do it when significant time changes, for instance: timezone updated.

## How is it done?
Forwarding call in `AppDelegate` to `TimeService`

### Why not another way?
I used to fetch the timeone info inside `TimeService` every `tick`, but that's not very performant. Let the system handle it is probably the easiest way.

### Side effects
🤷🏻‍♂️

## Review hints
- Change the timezone when the app is running
- Check whether reload is called

## :squid: Permissions
🐙

### CODE REVIEW

File: Toggl.Foundation/TimeService.cs [hunk 1]
```diff
@@ -52,6 +57,11 @@ public Task RunAfterDelay(TimeSpan delay, Action action)
                 .ToTask();
         }
 
+        public void SignificantTimeChanged()
```
Comment (juanlaube) [inline Toggl.Foundation/TimeService.cs]: `=>` this method!

File: Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs [hunk 2]
```diff
@@ -248,7 +248,12 @@ public override async Task Initialize()
                 .DisposedBy(disposeBag);
 
             timeService.MidnightObservable
-                .Subscribe(onMidnight)
+                .SelectUnit()
+                .Subscribe(onSignificantTimeChange)
+                .DisposedBy(disposeBag);
+
+            timeService.SignificantTimeChangeObservable
+                .Subscribe(onSignificantTimeChange)
```
Comment (juanlaube) [inline Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs]: `.VoidSubscribe` here and you can ignore the `Unit` param in `onSignificantTimeChange`

File: Toggl.Daneel/Startup/AppDelegate.cs [hunk 3]
```diff
@@ -137,6 +140,11 @@ public override void PerformActionForShortcutItem(UIApplication application, UIA
             }
         }
 
+        public override void ApplicationSignificantTimeChange(UIApplication application)
```
Comment (juanlaube) [inline Toggl.Daneel/Startup/AppDelegate.cs]: `=>` this method!

Comment (willsb) [reply Toggl.Daneel/Startup/AppDelegate.cs]: Actually, void methods should not use EBM :P

File: Toggl.Daneel/ViewControllers/MainViewController.cs [hunk 4]
```diff
@@ -210,6 +210,10 @@ public override void ViewDidLoad()
                 SendFeedbackSuccessView.Rx().AnimatedIsVisible());
             this.BindVoid(SendFeedbackSuccessView.Rx().Tap(), ViewModel.RatingViewModel.CloseFeedbackSuccessView);
 
+            ViewModel.ShouldReloadTELog
```
Comment (willsb) [inline Toggl.Daneel/ViewControllers/MainViewController.cs]: Properly named properties do not increase the binary size. Don't write `TE`, write `TimeEntry`

**Pull request merged**